asm/tkmk00_decode.s

```diff
@@ -10,6 +10,20 @@
 
 .section .text, "ax"
 
+/*
+See these links for potential references to implement our own TKMK00 (de)compression code
+https://github.com/mib-f8sm9c/MiscellaneousHacks/blob/master/BetaTools/MarioKartTestingTool/TKMK00Encoder.cs
+https://skelux.net/showthread.php?tid=47&page=8
+
+This link appears to have code near identical to libtkmk000
+https://github.com/mib-f8sm9c/MiscellaneousHacks/blob/master/BetaTools/MarioKartTestingTool/TKMKTest.cs
+Did libtkmk00 pull from that link, or vice-versa?
+
+Maybe tkmk00 wasn't compiled with IDO. Its explicit usage of registers rather than variables is something that IDO can't do (I think?) but GCC can.
+Or mabybe its C++ code and that interacts with registers differently?
+Or maybe its all hand written?
```

Comment:
It's my understanding that this assembly was hand-written.

Reply:
It is if you know where to look.

Reply:
The code was likely assembled using IDO, however the code is almost certainly hand-written given the bal, add, and sub, instructions which are not used by any known compiler.